### DISCUSSION TITLE
yasnippet  insert citation

### DISCUSSION THREAD

**Discussion by fab6** [emacs-citar/citar#726] [General]
Hi

 I am trying to write a small yasnippet snippet to insert selected keys:

```
# -*- mode: snippet -*-
# contributor: fb
# key: cite
# group: Bibtex
# name: citar-insert
# --
`(citar-insert-citation CITEKEYS)`
```
This does not work for me as I want to select the CITEKEYS before.
Does anyone have an idea how to get this running?

Best regards
Fab

**Comment by bdarcus** [comment]
Perhaps you should start from the beginning: what are you trying to achieve, in which major-mode(s)?

We do have `citar-insert-keys`.

**Comment by fab6** [comment]
sorry, I would like to type e.g. `cite` as a snippet, which calls `citar-insert-citation` in org- and latex-mode. Usually I run `citar-insert-citation` with a keybinding, which then opens the selection for existing bibtex items. So I would just shift the call for the function to a snippet.

**Reply by bdarcus** [reply]
I don't know myself.

I know it's not really documented (still needs some work), but have you tried  `citar-capf `?

**Reply by fab6** [reply]
ok, no problem, no, did not try yet `citar-capf`
Thank you for your quick help
I will see if I can figure it out

**Reply by bdarcus** [reply]
I recently updated the capf, and documented it:

https://github.com/emacs-citar/citar#citar-capf

**Reply by fbraennstroem** [reply]
thank you!
